x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/AllocationRoutedStep.java

```diff
@@ -68,42 +61,35 @@ public Result isConditionMet(Index index, ClusterState clusterState) {
         // if the allocation has happened
         RoutingAllocation allocation = new RoutingAllocation(ALLOCATION_DECIDERS, clusterState.getRoutingNodes(), clusterState, null,
                 System.nanoTime());
+
         int allocationPendingAllShards = 0;
 
         ImmutableOpenIntMap<IndexShardRoutingTable> allShards = clusterState.getRoutingTable().index(index).getShards();
         for (ObjectCursor<IndexShardRoutingTable> shardRoutingTable : allShards.values()) {
-            int allocationPendingThisShard = 0;
-            int shardCopiesThisShard = shardRoutingTable.value.size();
             for (ShardRouting shardRouting : shardRoutingTable.value.shards()) {
                 String currentNodeId = shardRouting.currentNodeId();
                 boolean canRemainOnCurrentNode = ALLOCATION_DECIDERS
                         .canRemain(shardRouting, clusterState.getRoutingNodes().node(currentNodeId), allocation)
                         .type() == Decision.Type.YES;
                 if (canRemainOnCurrentNode == false) {
```

Comment:
We need to add a check for the shard being started here too otherwise we will end up with the same issue on the allocate action but it will be harder to diagnose

Reply:
I'd like to address that in a separate PR, since I think it warrants more testing for the full ramifications, would that be okay with you?

Reply:
ok, could you create an issue so we can track the extra work and it doesn't get missed?

Reply:
Certainly, opened #35258